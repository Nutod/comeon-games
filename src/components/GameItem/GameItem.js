import React, { memo } from "react";
import PropTypes from "prop-types";

function GameItem({ name, description, icon, code }) {
  return (
    <div className="game item">
      <div className="ui small image">
        <img src={icon} alt="game-icon" />
      </div>
      <div className="content">
        <div className="header">
          <b className="name">{name}</b>
        </div>
        <div className="description">
          <p>{description}</p>
        </div>
        <div className="extra">
          <div className="play ui right floated secondary button inverted">
            Play
            <i className="right chevron icon" />
          </div>
        </div>
      </div>
    </div>
  );
}

GameItem.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired
};

export default memo(GameItem);
