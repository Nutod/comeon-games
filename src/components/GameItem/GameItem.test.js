import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import GameItem from "./GameItem";

describe("<GameItem /> Snapshot", () => {
  const wrapper = shallow(<GameItem />);

  it("Should match snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
