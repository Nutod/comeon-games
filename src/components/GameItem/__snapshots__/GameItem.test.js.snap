// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameItem /> Snapshot Should match snapshot 1`] = `
<div
  className="game item"
>
  <div
    className="ui small image"
  >
    <img
      alt="game-icon"
    />
  </div>
  <div
    className="content"
  >
    <div
      className="header"
    >
      <b
        className="name"
      />
    </div>
    <div
      className="description"
    >
      <p />
    </div>
    <div
      className="extra"
    >
      <Link
        to={
          Object {
            "pathname": "/undefined",
          }
        }
      >
        <div
          className="play ui right floated secondary button inverted"
        >
          Play
          <i
            className="right chevron icon"
          />
        </div>
      </Link>
    </div>
  </div>
</div>
`;
